curriculum/js1/8.js

```diff
@@ -5,8 +5,18 @@
  * @returns {number}
  */
 
-const solution = (a, b) => {
-  return 0;
+const solution = (a, b, n=2, result=1) => {
+  if ((a === 1) || (b === 1))
+    return result;
+
+  const min = a > b? b : a;
+
+  if (min < n)
+    return result;
+
+  return (!(a % n) && (!(b % n)))?
+    solution(a/n, b/n, n, n*result) :
+    solution(a, b, n+1, result);
```

Comment:
really inefficient solution. Think.... 1 is definitely a greatest common denominator right? start result as 1, counter as 1, and then if a%counter === 0 && b & counter == 0 then result = counter

Reply:
Sorry Song, I did not follow the hint.  Do you mean:

if (((a%counter) === 0) && ((b%counter) === 0))
  result = counter

then increment the counter and recurse?